exporter/exporterhelper/profiles.go

```diff
@@ -0,0 +1,161 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterhelper // import "go.opentelemetry.io/collector/exporter/exporterhelper"
+
+import (
+	"context"
+	"errors"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentprofiles"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/consumer/consumererror/consumererrorprofiles"
+	"go.opentelemetry.io/collector/consumer/consumerprofiles"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterprofiles"
+	"go.opentelemetry.io/collector/exporter/exporterqueue"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+)
+
+var profilesMarshaler = &pprofile.ProtoMarshaler{}
+var profilesUnmarshaler = &pprofile.ProtoUnmarshaler{}
+
+type profilesRequest struct {
+	pd     pprofile.Profiles
+	pusher consumerprofiles.ConsumeProfilesFunc
+}
+
+func newProfilesRequest(pd pprofile.Profiles, pusher consumerprofiles.ConsumeProfilesFunc) Request {
+	return &profilesRequest{
+		pd:     pd,
+		pusher: pusher,
+	}
+}
+
+func newProfileRequestUnmarshalerFunc(pusher consumerprofiles.ConsumeProfilesFunc) exporterqueue.Unmarshaler[Request] {
+	return func(bytes []byte) (Request, error) {
+		profiles, err := profilesUnmarshaler.UnmarshalProfiles(bytes)
+		if err != nil {
+			return nil, err
+		}
+		return newProfilesRequest(profiles, pusher), nil
+	}
+}
+
+func profilesRequestMarshaler(req Request) ([]byte, error) {
+	return profilesMarshaler.MarshalProfiles(req.(*profilesRequest).pd)
+}
+
+func (req *profilesRequest) OnError(err error) Request {
+	var profileError consumererrorprofiles.Profiles
+	if errors.As(err, &profileError) {
+		return newProfilesRequest(profileError.Data(), req.pusher)
+	}
+	return req
+}
+
+func (req *profilesRequest) Export(ctx context.Context) error {
+	return req.pusher(ctx, req.pd)
+}
+
+func (req *profilesRequest) ItemsCount() int {
+	return req.pd.SampleCount()
+}
+
+type profileExporter struct {
+	*baseExporter
+	consumerprofiles.Profiles
+}
+
+// NewProfilesExporter creates an exporterprofiles.Profiless that records observability metrics and wraps every request with a Span.
+func NewProfilesExporter(
```

Comment:
We will eventually have to move this away to a different package when we mark exporterhelper as 1.0. We can do this now on this PR or I can do it later sicne there is other experimental API

Reply:
Done in https://github.com/open-telemetry/opentelemetry-collector/pull/11131/commits/c14ae5aba347341a7c1f854ad4b239a96f26b38f.